### PR TITLE
fix: universal macOS release support

### LINKED ISSUE #13: Darwin asset missing
CMake 3.19.2 comes in a universal build now, but it causes lookups like the one in this action to fail. (Also see scikit-build/cmake-python-distributions#123 - it broke the update script there too). If you look here: https://api.github.com/repos/Kitware/CMake/releases/tags/v3.19.2, the new name is `cmake-3.19.2-macos-universal`.

### PR DESCRIPTION
**Pull request by henryiii**
Trying to fix #13

### CODE REVIEW

File: src/version.ts [hunk 1]
```diff
@@ -54,6 +54,8 @@ function extractFileTypeFrom(filename: string): string {
 function extractArchFrom(filename: string): string {
   if (filename.match(/x86_64/)) {
     return 'x86_64';
+  } else if (filename.match(/universal/) {
```
Comment (henryiii) [inline src/version.ts:57]:
```suggestion
  } else if (filename.match(/universal/)) {
```

**Comment by henryiii** [comment]
Well, you can take this as a starting point, perhaps? Not obvious to me why it didn't fix the problem.

**Comment by jwlawson** [comment]
Thanks for looking into this. I think your changes are sufficient to fix the problem. I've built on them in #15 and added a few tests as well.

This is completely not clear, but the way this action works you need to compile any Typescript changes into dist/index.js for them to take effect. This should be as simple as running `npm run prepare` and committing the changes. Roughly this converts the Typescript into minified javascript. The github action loads directly from dist/index.js and so doesn't need to know anything about the Typescript itself.

I should probably add some notes on how to contribute to the readme at some point.

**Comment by henryiii** [comment]
Ahh! That makes much more sense. I could not see why that failed to fix the problem - a prepare step would explain it. Maybe you could add a GHA that runs prepare and shows a helpful error if git diff fails? (maybe you could even make it a pre-commit step, and pre-commit users would get it automatically, too, besides just the online check...) Just thinking about what I'd do.

If you want to keep working on the same PR, you can just use `gh pr checkout 14` with the GitHub CL client and that handles all the remote business for you, FYI. Even if the author don't make a new branch name, which I used to hate messing with. Perfectly fine this way too, of course. :)

Thanks!

**Pull request closed without merging**